src/rez/developer_package.py

```diff
@@ -159,28 +168,36 @@ def _validate_includes(self):
     def _get_preprocessed(self, data):
         """
         Returns:
-            (DeveloperPackage, new_data) 2-tuple IFF the preprocess function
+            (DeveloperPackage, new_data) 2-tuple IF the preprocess function
             changed the package; otherwise None.
         """
         from rez.serialise import process_python_objects
         from rez.utils.data_utils import get_dict_diff_str
         from copy import deepcopy
 
-        with add_sys_paths(config.package_definition_build_python_paths):
-            preprocess_func = getattr(self, "preprocess", None)
+        package_preprocess_mode = self.config.package_preprocess_mode
 
-            if preprocess_func:
-                print_info("Applying preprocess from package.py")
+        def _get_package_level():
+            print_info("Applying preprocess from package.py")
```

Comment:
Message printed too early? I would also say '...from package definition' rather than package.py (you can configure rez to use a different filename).